heron/spi/src/java/com/twitter/heron/spi/statemgr/SchedulerStateManagerAdaptor.java

```diff
@@ -179,6 +189,63 @@ public Boolean deleteSchedulerLocation(String topologyName) {
     return awaitResult(delegate.getSchedulerLocation(null, topologyName));
   }
 
+  /**
+   * Checks to see if the execution state exists
+   *
+   * @return a boolean indicating whether the execution state exists or not
+   */
+  public boolean doesExecutionStateExist(String topologyName) {
+    return awaitResult(delegate.doesExecutionStateExists(topologyName));
+  }
+
+  /**
+   * Clean all states of a heron topology
+   * 1. Topology def and ExecutionState are required to exist to delete
+   * 2. TMasterLocation, SchedulerLocation and PhysicalPlan may not exist to delete
+   */
+  public Boolean cleanState(String topologyName) {
+    LOG.fine("Cleaning up topology state");
+
+    Boolean result;
+
+    // It is possible that  TMasterLocation, PhysicalPlan and SchedulerLocation are not set
+    // Just log but don't consider them failure
+    result = deleteTMasterLocation(topologyName);
+    if (result == null || !result) {
+      // We would not return false since it is possible that TMaster didn't write physical plan
+      LOG.warning("Failed to clear TMaster location. Check whether TMaster set it correctly.");
+    }
+
+    result = deletePhysicalPlan(topologyName);
+    if (result == null || !result) {
+      // We would not return false since it is possible that TMaster didn't write physical plan
```

Comment:
please remove this and the 2 other (previously) copy/pasted comments that are erroneous.